common/ManagementTestShared/Redesign/InherentCheckTests.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Azure.ResourceManager.TestFramework
+{
+    public sealed class InherentCheckTests
```

Comment:
Should it be `InheritanceCheckTests` or `InheritCheckTests`?
The name of this class actually could be more generic, we could put all this kind of reflection based test cases in this class.

Reply:
Maybe if we make this class `partial`, we could directly assign values to the `ExceptionList` property?
Could you have a try?

Reply:
Solved